int16 x_min 
int16 x_max
int16 cls
float64[] probability_distribution
